### ISSUE TITLE
Error epub

### ISSUE THREAD

**Issue by TonyDiana** [sphinx-doc/sphinx#3657]
Subject: <what happen when you do on which document project>

### Problem
No sure

#### Procedure to reproduce the problem
```
<Paste your command-line here which cause the problem>
```

#### Error logs / results
```
# Sphinx version: 1.5.5
# Python version: 2.7.12 (CPython)
# Docutils version: 0.13.1 release
# Jinja2 version: 2.9.6
# Last messages:
#   writing output... [ 25%] api/serv_soft/gui
#   writing output... [ 27%] api/serv_soft/ini
#   writing output... [ 29%] api/serv_soft/instala
#   writing output... [ 31%] api/serv_soft/m360
#   writing output... [ 34%] api/serv_soft/problemas
#   writing output... [ 36%] api/serv_soft/python
#   writing output... [ 38%] api/serv_soft/rec
#   writing output... [ 40%] api/serv_soft/so
#   writing output... [ 43%] api/solamente
#   writing output... [ 45%] genindex
# Loaded extensions:
#   sphinx.ext.todo (1.5.5) from /home/tony/.local/lib/python2.7/site-packages/sphinx/ext/todo.pyc
#   sphinx.ext.autodoc (1.5.5) from /home/tony/.local/lib/python2.7/site-packages/sphinx/ext/autodoc.pyc
#   alabaster (0.7.10) from /home/tony/.local/lib/python2.7/site-packages/alabaster/__init__.pyc
Traceback (most recent call last):
  File "/home/tony/.local/lib/python2.7/site-packages/sphinx/cmdline.py", line 296, in main
    app.build(opts.force_all, filenames)
  File "/home/tony/.local/lib/python2.7/site-packages/sphinx/application.py", line 333, in build
    self.builder.build_update()
  File "/home/tony/.local/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 251, in build_update
    'out of date' % len(to_build))
  File "/home/tony/.local/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 322, in build
    self.write(docnames, list(updated_docnames), method)
  File "/home/tony/.local/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 360, in write
    self._write_serial(sorted(docnames), warnings)
  File "/home/tony/.local/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 368, in _write_serial
    self.write_doc(docname, doctree)
  File "/home/tony/.local/lib/python2.7/site-packages/sphinx/builders/epub.py", line 416, in write_doc
    StandaloneHTMLBuilder.write_doc(self, docname, doctree)
  File "/home/tony/.local/lib/python2.7/site-packages/sphinx/builders/html.py", line 461, in write_doc
    self.handle_page(docname, ctx, event_arg=doctree)
  File "/home/tony/.local/lib/python2.7/site-packages/sphinx/builders/epub.py", line 506, in handle_page
    self.fix_genindex(addctx['genindexentries'])
KeyError: 'genindexentries'
```
- <public link of unexpected result if you have>

#### Expected results
<Describe what to actually do>

### Reproducible project / your project
- <link to your project, or attach zipped small project sample>

### Environment info
- OS: <Unix/Linux/Mac/Win/other with version>
- Python version:
- Sphinx version:
- <Extra tools e.g.: Browser, tex or something else>

**Comment by shibukawa** [comment]
May I have any sample document to reproduce this issue?

**Comment by TonyDiana** [comment]
The document is immense, as about 118 elements, 4.2 MB HTML in total, plus the docstring python. It works in html, that is the error message that gives

**Comment by tk0miya** [comment]
I reproduced this problem. A document starts with `genindex` causes this error.
```
$ cat $EOF>>genindex.rst
genindex
========
$ make epub
...
writing output... [ 60%] genindex
Traceback (most recent call last):
  File "/Users/tkomiya/work/sphinx/sphinx/cmdline.py", line 296, in main
    app.build(opts.force_all, filenames)
  File "/Users/tkomiya/work/sphinx/sphinx/application.py", line 333, in build
    self.builder.build_update()
  File "/Users/tkomiya/work/sphinx/sphinx/builders/__init__.py", line 251, in build_update
    'out of date' % len(to_build))
  File "/Users/tkomiya/work/sphinx/sphinx/builders/__init__.py", line 322, in build
    self.write(docnames, list(updated_docnames), method)
  File "/Users/tkomiya/work/sphinx/sphinx/builders/__init__.py", line 360, in write
    self._write_serial(sorted(docnames), warnings)
  File "/Users/tkomiya/work/sphinx/sphinx/builders/__init__.py", line 368, in _write_serial
    self.write_doc(docname, doctree)
  File "/Users/tkomiya/work/sphinx/sphinx/builders/epub.py", line 416, in write_doc
    StandaloneHTMLBuilder.write_doc(self, docname, doctree)
  File "/Users/tkomiya/work/sphinx/sphinx/builders/html.py", line 461, in write_doc
    self.handle_page(docname, ctx, event_arg=doctree)
  File "/Users/tkomiya/work/sphinx/sphinx/builders/epub.py", line 506, in handle_page
    self.fix_genindex(addctx['genindexentries'])
KeyError: 'genindexentries'
```

**Comment by TonyDiana** [comment]
Yes, that's exactly it, thanks

**Comment by tk0miya** [comment]
Reopened, I think this should be fixed.

**Comment by TonyDiana** [comment]
Yes :), tx

**Comment by tk0miya** [comment]
Fixed at #3675 
Thank you for reporting!
